The GLORIOUS Supplypod Renaissance PART 2: The COOL GRAPHICS Update [EPIC MEME VIDEO INSIDE]

MrDoomBringer opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->
# [Please check out this EPIC OFFICIAL PR TRAILER before reading (very important)](https://youtu.be/4RckVdx20oQ)
the video is kinda autistic

dont forget to upvote
## About The Pull Request


All sprites and epic meme videos made with love by yours truly
<sup>Except for the 2 missile sprites. Those were taken and modified from NSV13 who in turn ported them from TGMC</sup>
![nupod_all_decals](https://user-images.githubusercontent.com/29008542/85791031-2f9dae00-b6ff-11ea-922e-03bcea3be88b.png)



Anyways, this adds:
New pod, reverse-pod, and target indicator sprites
Pod-ground collision effects (impact craters basically)
Cool looking vapor trails a la the half-life 2 headcrab shells 

And also:
Fixes seethrough pods not actually working
Makes reverse-mode a bit more predictable
Has some minor code improvements with how supplypod-reverse-mode works
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Yall really let #33436 get merged with the pods looking like that?
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:  MrDoomBringer
imageadd: Nanotrasen's Transportation and Logistics Subdivision has spent the better part of their budget developing a brand new model of Supplypod. Internal investigations have revealed the only real differences are almost purely aesthetic.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

MrDoomBringer commented:
Note that supplypod code has gotten pretty bloated, next time on Part 3 we'll refactor all this nastiness.

optimumtact commented:
I swear to god every time you make a PR your powerlevel increases

kopoba commented:
can we have sindi red and clown pink?

MrDoomBringer commented:
@kopoba garner some upthumbs and i'll make it so

zxaber commented:
>Fixes seethrough pods not actually working

Wait, what? I use them all the time and they seem fine enough. What was the issue?

PKPenguin321 commented:
you should resprite the gondola pod so the opened door is actually its open mouth with its lower jaw dropping really really low

MrDoomBringer commented:
@zxaber From what I could tell they didn't seem to work properly with mobs

zxaber commented:
Oh, I suppose I haven't really tried with mobs. If that wasn't working, good catch.

GravitronBlue commented:
I'd like to see the bluespace pod keep the white paint it had previously.

MrDoomBringer commented:
@GravitronBlue I tried experimenting with that for a while but it never came out quite right, I'll give it another stab in a bit and post it here to see if you like it

Doosteroni commented:
There should be like a skeleton in the blood cult pod

Ryll-Ryll commented:
Hypest trailer 2020

Dorsisdwarf commented:
I agree that I'm not sure if I like the loss of all color differentiation leaving all standard supply pods as just "choose a decal" on identical bases, but these sprites are VERY nice.

MrDoomBringer commented:
Thoughts on something like this?
![image](https://user-images.githubusercontent.com/29008542/85888139-ffb1e180-b7b6-11ea-9acf-770ab0eb3404.png)
Shoutout to @davethwave for the syndicate pod sprite!

GoldenAlpharex commented:
That looks amazing! Just one thing, though: Would it possible to also have the old sprites still available? That way, we could also still mess around with those too, for people that liked what they looked like. Otherwise, good job, this is just insane.

MrDoomBringer commented:
@GoldenAlpharex whew why not

Bastian0930 commented:
litterally the video made me laugh

please make more over the top videos and ping me when done with them

it's fucking hilarious

SweptWasTaken commented:
Did you spend more time on the video or the pr?

(I approve of both, well done)

Fikou commented on code/__DEFINES/cargo.dm:
```diff
@@ -13,23 +13,30 @@
+    list(TRUE, "missile",			RUBBLE_THIN,	"cruise missile", 						"A big ass missile that didn't seem to fully detonate. It was likely launched from some far-off deep space missile silo. There appears to be an auxillery payload hatch on the side, though manually opening it is likely impossible."),\
+    list(TRUE, "syndicatemissile",	RUBBLE_THIN,	"\improper Syndicate cruise missile", 	"A big ass, blood-red missile that didn't seem to fully detonate. It was likely launched from some deep space Syndicate missile silo. There appears to be an auxillery payload hatch on the side, though manually opening it is likely impossible."),\
+    list(TRUE, "box",				RUBBLE_WIDE,	"\improper Aussec supply crate", 		"An incredibly sturdy supply crate, designed to withstand orbital re-entry. Has 'Aussec Armory - 2532' engraved on the side."),\
+    list(TRUE, "clown", "clown",	RUBBLE_NORMAL,	"\improper HONK pod", 				"A brightly-colored supply pod. It likely originated from the Clown Federation."),\
```
you put in "clown" twice

MrDoomBringer commented:
![image](https://user-images.githubusercontent.com/29008542/85923892-bb791c80-b85c-11ea-9174-219273382491.png)
slightly more hi-def base sprites also comin in

Moja10 commented:
MrDoomBringer is the new oranges

MrHorizons commented:
> MrDoomBringer is the new oranges

I'll believe it when he removes cloning again.

MrDoomBringer commented:
![image](https://i.imgur.com/icYjgt9.gif)
cool new effects booster pack just added

MemedHams commented:
Quality animation in YOUR spessman game?

It's more likely than you think.

kriskog reviewed (dismissed):
Fails the linter.
```
file ./code/modules/cargo/supplypod.dm is missing a trailing newline
```

GoldenAlpharex commented:
Also, I don't know if that issue has been addressed with this PR, but just to be sure: Do you think you could remove the Alert Ghost option from being turned on by default? It's kind of annoying, as we almost never actually want it on.

kriskog commented:
>  It's kind of annoying, as we almost never actually want it on.

Why would you not want this on? Its there to entertain bored ghosts.

GoldenAlpharex commented:
> > It's kind of annoying, as we almost never actually want it on.
> 
> Why would you not want this on? Its there to entertain bored ghosts.

Because I mostly use it for uninteresting stuff.

MrDoomBringer commented:
@GoldenAlpharex Soon(tm) i plan to add save slots (of which you can choose a default) for the supplypod menu, so that should solve your issue

MrDoomBringer commented:
linter has been quelled

MrDoomBringer committed:
first pass


WIP overlays


icon upd8


more


need to figure out whats going on with extractionpods


compile


the icon update


more!!


why are skillcapes broken


boat lmao


seethrough fix


Second drive-by


Code cleanup and improvements

Specifically surrounding contractor pods and reverse mode working properly
accidently left in an extra dmi whups


do the impossible see the invisible


new effect booster pack


MFW MRW Linter fail

MrDoomBringer committed:
reverse sprite fix and no more message_admin()s

MrDoomBringer committed:
fixes rubble a bit

MrDoomBringer committed:
fix reverse icon

yeeyeh commented:
The trailer alone is good enough to sell me on this. 10/10 beautiful execution

Twaticus reviewed (approved):
:heart:
